### PR TITLE
Implements setting and modifying a map/layers

### PR DESCRIPTION
**Pull request by puneet-pdx**
- set a map on a composable mapview and render it 
  using `MapState(arcGISMap: ArcGISMap? = null)`
- change a map on a composable mapview and ensure that the rendered map is updated 
  using setter `setArcGISMap(arcGISMap: ArcGISMap)` 
- add/remove layers on a map already rendered by a composable mapview 
  using `mapState.arcGISMap.value?.operationalLayers`

### CODE REVIEW

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/GeoComposeState.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+package com.arcgismaps.toolkit.geocompose
+
+/**
+ * Represents the state for the GeoCompose.
+ *
+ * @since 200.3.0
+ */
+public sealed interface GeoComposeState {
```
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/GeoComposeState.kt:25]: updated

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/GeoComposeState.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    // TODO add implementation
+}
+
+public fun GeoComposeState(): GeoComposeState =
+    GeoComposeStateImpl()
+
+public open class GeoComposeStateImpl() : GeoComposeState {
```
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/GeoComposeState.kt:32]: removed

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/GeoComposeState.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * Represents the state for the GeoCompose.
+ *
+ * @since 200.3.0
+ */
+public sealed interface GeoComposeState {
+    // TODO add implementation
+}
+
+public fun GeoComposeState(): GeoComposeState =
+    GeoComposeStateImpl()
```
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/GeoComposeState.kt:30]: No longer needed.

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/Map.kt [hunk 4]
```diff
@@ -18,14 +17,38 @@
...
 @Composable
-public fun Map() {
-    // Todo implementation...
-    Text("Implement Map here! ")
+public fun Map(modifier: Modifier = Modifier, mapState: MapState = MapState()) {
+
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val context = LocalContext.current
+    val mapView = remember { MapView(context) }
+    val map by mapState.arcGISMap.collectAsState()
```
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/Map.kt:38]: updated

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapState.kt [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+/**
+ * Represents the state for the Map.
+ *
+ * @since 200.3.0
+ */
+public sealed interface MapState : GeoComposeState {
```
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapState.kt:30]: updated

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapState.kt [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    public val arcGISMap: StateFlow<ArcGISMap?>
+
+    public fun setArcGISMap(arcGISMap: ArcGISMap)
```
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapState.kt:33]: As per our discussion I have left it in for now.

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapState.kt [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+public fun MapState(arcGISMap: ArcGISMap? = null): MapState = MapStateImpl(arcGISMap)
```
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapState.kt:36]: I have removed this as we are not implementing this pattern based on your comment.

**Review by gunt0001 — APPROVED** [review]
@puneet-pdx - looks good 👍

**Review by shubham7109 — APPROVED** [review]
Modifying map layers looks good. Just a minor comment:

**Pull request merged**